Use torch.set_default_dtype in test_data_parallel and rename dtype2prec

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32962 Use torch.set_default_dtype in test_data_parallel and rename dtype2prec**

As per @gchanan's comments on
https://github.com/pytorch/pytorch/pull/30445, I've used
`torch.set_default_dtype` in test_data_parallel instead of specifying
dtype=torch.double everywhere. Also, renamed dtype2prec to dtype2prec_DONTUSE

Differential Revision: [D19714374](https://our.internmc.facebook.com/intern/diff/D19714374/)

pritamdamania committed:
Use torch.set_default_dtype in test_data_parallel and rename dtype2prec

As per @gchanan's comments on
https://github.com/pytorch/pytorch/pull/30445, I've used
`torch.set_default_dtype` in test_data_parallel instead of specifying
dtype=torch.double everywhere. Also, renamed dtype2prec to dtype2prec_DONTUSE

Differential Revision: [D19714374](https://our.internmc.facebook.com/intern/diff/D19714374/)

[ghstack-poisoned]

gchanan commented on torch/testing/_internal/common_utils.py:
```diff
@@ -1462,7 +1462,8 @@ def _assertGradAndGradgradChecks(test_case, apply_fn, inputs):
 
-dtype2prec = {torch.float: 1e-5,
+# Using @precisionOverride is the recommended way instead of this.
+dtype2prec_DONTUSE = {torch.float: 1e-5,
```
can you fix the indent

gchanan commented on torch/testing/_internal/common_utils.py:
```diff
@@ -1462,7 +1462,8 @@ def _assertGradAndGradgradChecks(test_case, apply_fn, inputs):
 
 
-dtype2prec = {torch.float: 1e-5,
+# Using @precisionOverride is the recommended way instead of this.
```
maybe add a bit more context: using precisionOverride specific to your test is the recommended way of doing this.  These are just some values that worked for test_nn.

gchanan reviewed (approved):
address the comments then looks good to go.

pritamdamania committed:
Update on "Use torch.set_default_dtype in test_data_parallel and rename dtype2prec"


As per @gchanan's comments on
https://github.com/pytorch/pytorch/pull/30445, I've used
`torch.set_default_dtype` in test_data_parallel instead of specifying
dtype=torch.double everywhere. Also, renamed dtype2prec to dtype2prec_DONTUSE

Differential Revision: [D19714374](https://our.internmc.facebook.com/intern/diff/D19714374/)

[ghstack-poisoned]
